### PR TITLE
allow to disable magithub by default

### PR DESCRIPTION
**Pull request by ericdanan**
Currently, magithub is enabled by default for all repositories, unless I explicitly disable for each repo. This PR simply introduces a custom variable `magithub-enabled-by-default` allowing to disable magithub by default, and modifies `magithub-enabled-p` to take it into account.

### CODE REVIEW

File: magithub-core.el [hunk 1]
```diff
@@ -104,9 +104,17 @@ created automatically."
...
 
+(defcustom magithub-enabled-by-default t
+  "Whether Magithub is enabled by default."
+  :group 'magithub
+  :type 'boolean)
+
 (defun magithub-enabled-p ()
   "Returns non-nil when Magithub is enabled for this repository."
-  (and (member (magit-get "magithub" "enabled") '("yes" nil)) t))
+  (let ((enabled (magit-get "magithub" "enabled")))
```
Comment (vermiculus) [inline magithub-core.el]: This might make more sense readability-wise as a full `if` form.

File: magithub-core.el [hunk 2]
```diff
@@ -104,9 +104,17 @@ created automatically."
     (magit-refresh))
   (message "Magithub is now disabled in this repository"))
 
+(defcustom magithub-enabled-by-default t
+  "Whether Magithub is enabled by default."
```
Comment (vermiculus) [inline magithub-core.el]: Usually boolean-type docstrings are of the form 'non-nil if …'; can you make this change?

File: magithub-core.el (hunk 1 above)
Comment (ericdanan) [reply magithub-core.el]: Sorry can you explain to me what you mean more precisely?

Comment (vermiculus) [reply magithub-core.el]: The logic is just a little hard to follow as it stands. I'm not sure what kind of expression I was thinking that involved `if`; I should've written it down ☹️

Maybe something using `cond`:

```elisp
(cond
 ((string= enabled "yes") t)
 ((string= enabled "no") nil)
 (t magithub-enabled-by-default))
```

**Comment by ericdanan** [comment]
OK I did the changes, rebased, and then force-pushed to my fork. Hope I didn't mess up anything.

**Comment by vermiculus** [comment]
Nope, you're all good! Thanks!
